hbase-server/src/test/java/org/apache/hadoop/hbase/HBaseTestingUtility.java

```diff
@@ -213,6 +213,10 @@
 
   private volatile AsyncClusterConnection asyncConnection;
 
+  // Tracks any other connections created with custom  client config. Used for testing clients with custom
+  // configurations. Tracked here so that they can be cleaned up on close() / restart.
+  private List<AsyncClusterConnection> customConnections = Collections.synchronizedList(new ArrayList<>());
+
```

Comment:
I didn't quite follow why you needed to define these extra connections. Are you using both types of registry implementation on a single util instance?

Reply:
> because we need the service side meta/master tracking to happen using ZK and just the client side to use HMasterAsyncRegistry

Oh, didn't know that was the intention. I thought it was just fine to leave even server side processes acting as client to use the master registry. Is the concern here related to an rpc overload?

Reply:
I have the same question. Only the masters need to do things differently. All other clients, including embedded server side clients, should use the new service? So we get a nice common interposition point between all clients and the source of truth. Except the masters themselves, of course, which need to go directly to the source of truth.

Reply:
@apurtell Agree, that makes the whole design simpler. That is my end goal.